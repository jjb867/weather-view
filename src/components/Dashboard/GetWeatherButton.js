import React from 'react';

const GetWeatherButton = props => {
    return (
        <button className="button is-link is-medium" onClick={props.onClick}>
            Get Weather
        </button>
    );
};

export default GetWeatherButton;
