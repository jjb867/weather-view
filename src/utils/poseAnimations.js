import posed from 'react-pose';

export const Container = posed.div({
    enter: { staggerChildren: 50 }
});

export const Paragraph = posed.p({
    enter: {x:20, opacity: 1 },
    exit: {x:0, opacity: 0 }
});

export const GetWeatherAnimation = posed.div({
    hidden: { y: -10, x: -10, opacity: 0, transition: { duration: 250 } },
    visible: { y: 0, x: 0, opacity: 1, transition: { duration: 250 } },
});

export const charPoses = {
    exit: { opacity: 0, x: -20 },
    enter: { opacity: 1, x: 0 }
};

export const Input = posed.input({
    focusable: true,
    init: {
        scale: 1
    },
    focus: {
        scale: 1.0125
    }
});

export const Button = posed.div({
    hoverable: true,
    pressable: true,
    init: {
        scale: 1,
        boxShadow: '0px 0px 0px rgba(0,0,0,0)'
    },
    hover: {
        scale: 1.0125,
        boxShadow: '0px 2px 4px rgba(0,0,0,0.2)'
    },
    press: {
        scale: 1.06125,
        boxShadow: '0px 1px 2px rgba(0,0,0,0.1)'
    }
});

