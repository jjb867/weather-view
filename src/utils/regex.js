export const zipRegex = /^(\d{5})?$/;
export const cityRegex = /^[a-zA-Z]+(?:[\s-][a-zA-Z]+)*$/;