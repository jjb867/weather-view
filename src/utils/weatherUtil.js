import moment from 'moment';

export const convertKelvinToFahrenheit = kelvinTemp => {
    return parseInt(1.8 * (kelvinTemp - 273) + 32);
};

export const capitalizeFirstLetter = description => {
    return description.charAt(0).toUpperCase() + description.slice(1);
};

export const getCurrentTime = () => {
    return moment().format('h:mmA');
};

export const getWindDirection = angle => {
    let segments = 8;
    let degree = 360 / segments;
    angle = angle + degree / 2;

    if (angle >= 0 && angle < degree) {
        return 'N';
    } else if (angle >= degree && angle < 2 * degree) {
        return 'NE';
    } else if (angle >= 2 * degree && angle < 3 * degree) {
        return 'E';
    } else if (angle >= 3 * degree && angle < 4 * degree) {
        return 'SE';
    } else if (angle >= 4 * degree && angle < 5 * degree) {
        return 'S';
    } else if (angle >= 5 * degree && angle < 6 * degree) {
        return 'SW';
    } else if (angle >= 6 * degree && angle < 7 * degree) {
        return 'W';
    } else if (angle >= 7 * degree && angle < 8 * degree) {
        return 'NW';
    } else {
        return 'N';
    }
};

export const getLatestDescriptionIndex = weather => {
    let lastIndex = Object.keys(weather)[Object.keys(weather).length - 1];
    return parseInt(lastIndex);
};