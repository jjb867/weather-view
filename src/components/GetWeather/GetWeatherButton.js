import React from 'react';
import { Button } from '../../utils/poseAnimations';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const GetWeatherButton = props => {
    return (
        <Button className="button is-link is-medium" onClick={props.onClick}>
            <FontAwesomeIcon icon="cloud-sun-rain"/>&nbsp;
            Get Weather
        </Button>
    );
};

export default GetWeatherButton;
