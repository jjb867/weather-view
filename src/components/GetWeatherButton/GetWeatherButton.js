import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '../../utils/poseAnimations';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import classnames from 'classnames';

const GetWeatherButton = ({ handleOnClick, isProcessing }) => {
    return (
        <Button
            className={classnames('button is-link', {
                'is-loading': isProcessing,
            })}
            onClick={handleOnClick}
        >
            <FontAwesomeIcon icon="search" />
        </Button>
    );
};

GetWeatherButton.propTypes = {
    handleOnClick: PropTypes.func.isRequired,
    isProcessing: PropTypes.bool.isRequired,
};

export default GetWeatherButton;
