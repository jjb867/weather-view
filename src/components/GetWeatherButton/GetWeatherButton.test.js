import React from 'react';
import { shallow } from 'enzyme';
import GetWeatherButton from './GetWeatherButton';
import { Button } from '../../utils/poseAnimations';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

describe('<GetWeatherButton />', () => {
    let component;
    let handleOnClick;
    let isProcessing = false;
    let buttonWrapper;

    beforeAll(() => {
        handleOnClick = jest.fn();
        component = shallow(<GetWeatherButton isProcessing={isProcessing} handleOnClick={handleOnClick} />);
        buttonWrapper = component.find(Button);
    });

    it('matches snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('calls handleOnClick() on button click', () => {
        expect(component.find('.button.is-link')).toHaveLength(1);
        component.find('.button.is-link').simulate('click');
        expect(handleOnClick).toHaveBeenCalled();
    });

    it('has a Button animation element', () => {
        expect(buttonWrapper).toHaveLength(1);
    });

    it('renders the icon', () => {
        const icon = buttonWrapper.find(FontAwesomeIcon);
        expect(icon).toHaveLength(1);
        expect(icon.prop('icon')).toEqual('search');
    });
});
