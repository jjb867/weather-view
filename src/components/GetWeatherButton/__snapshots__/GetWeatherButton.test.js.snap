// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GetWeatherButton /> matches snapshot 1`] = `
<ForwardRef
  className="button is-link"
  onClick={[MockFunction]}
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="search"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    symbol={false}
    title=""
    transform={null}
  />
</ForwardRef>
`;
